functions/functionsv2/ocr/app/detect.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_ocr_detect]
+
+package ocr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"cloud.google.com/go/pubsub"
+	"golang.org/x/text/language"
+	visionpb "google.golang.org/genproto/googleapis/cloud/vision/v1"
+)
+
+// detectText detects the text in an image using the Google Vision API.
+func detectText(ctx context.Context, bucketName, fileName string) error {
+	log.Printf("Looking for text in image %v", fileName)
+	maxResults := 1
+	image := &visionpb.Image{
+		Source: &visionpb.ImageSource{
+			GcsImageUri: fmt.Sprintf("gs://%s/%s", bucketName, fileName),
+		},
+	}
+	annotations, err := visionClient.DetectTexts(ctx, image, &visionpb.ImageContext{}, maxResults)
```

Comment:
Not seeing where visionClient or translateClient is declared. Perhaps there's a piece of go magic I'm not understanding here.

Reply:
Oh, I see, all of these files are in the same directory so share a common namespace, and seeing that they are declared in the ocr_test.go file.